src/hapi/pipelines/database/humanitarian_needs.py

```diff
@@ -22,147 +23,112 @@ def __init__(
+            time_period = dataset.get_time_period()
+            time_period_start = time_period["startdate_str"]
+            time_period_end = time_period["enddate_str"]
+            resource = dataset.get_resource()
+            resource_id = resource["id"]
+            url = resource["url"]
+            headers, rows = reader.get_tabular_rows(url, dict_form=True)
+            # Admin 1 PCode,Admin 2 PCode,Sector,Gender,Age Group,Disabled,Population Group,Population,In Need,Targeted,Affected,Reached
+            for row in rows:
+                admin2_ref = self.get_admin2_ref(countryiso3, row)
+                if not admin2_ref:
+                    continue
+                population_group = row["Population Group"]
+                if population_group == "ALL":
+                    population_group = "*"
+                sector = row["Sector"]
+                sector_code = self._sector.get_sector_code(sector)
+                if not sector_code:
+                    logger.warning(f"Sector {sector} not found!")
+                    continue
+                gender = row["Gender"]
+                if gender == "a":
+                    gender = "*"
+                age_range = row["Age Range"]
+                min_age = row["Min Age"]
+                max_age = row["Max Age"]
+                disabled_marker = row["Disabled"]
+                if disabled_marker == "a":
+                    disabled_marker = "*"
+
+                def create_row(in_col, population_status):
+                    value = row[in_col]
+                    if value is None:
+                        return
+                    value = get_numeric_if_possible(value)
+                    if value < 0:
+                        logger.warning(
+                            f"Negative {population_status} {value}!"
                         )
-                        humanitarian_needs_row = DBHumanitarianNeeds(
-                            resource_hdx_id=resource_id,
-                            admin2_ref=self._admins.admin2_data[admin2_code],
-                            gender=gender,
-                            age_range=age_range,
-                            min_age=min_age,
-                            max_age=max_age,
-                            sector_code=sector_code,
-                            population_group=population_group,
-                            population_status=population_status,
-                            disabled_marker=disabled_marker,
-                            population=value,
-                            reference_period_start=time_period_start,
-                            reference_period_end=time_period_end,
+                        return
+                    if isinstance(value, float):
+                        logger.info(
+                            f"Rounding float {population_status} {value}!"
                         )
+                        value = round(value)
```

Comment:
Should we flag this to DPT?

Reply:
Maybe we can flag this to DPT in slack and remove the warning message, since there are so many rows it applies to.

Reply:
I've made a summary at the end of the run with this